lib/hmmlearn/hmm.py

```diff
@@ -307,18 +309,215 @@ def _do_mstep(self, stats):
                                      (cvweight + stats['post'][:, None, None]))
 
 
-_MULTINOMIALHMM_DOC_SUFFIX = """
+
+class MultinomialHMM(BaseHMM):
+    """
+    Hidden Markov Model with multinomial emissions
+
+    Attributes
+    ----------
+    n_features : int
+        Number of possible symbols emitted by the model (in the samples).
+
+    monitor_ : ConvergenceMonitor
+        Monitor object used to check the convergence of EM.
+
+    startprob_ : array, shape (n_components, )
+        Initial state occupation distribution.
+
+    transmat_ : array, shape (n_components, n_components)
+        Matrix of transition probabilities between states.
+
+    emissionprob_ : array, shape (n_components, n_features)
+        Probability of emitting a given symbol when in each state.
+
+    Examples
+    --------
+    >>> from hmmlearn.hmm import MultinomialHMM
+    """
+
+    def __init__(self, n_components, n_trials, n_features=2,
+                 startprob_prior=1.0, transmat_prior=1.0,
+                 algorithm="viterbi", random_state=None,
+                 n_iter=10, tol=1e-2, verbose=False,
+                 params="ste", init_params="ste",
+                 implementation="log"):
+        """
+        Parameters
+        ----------
+        n_components : int
+            Number of states, required
+            
+        n_trials : int
+            Number of trials, required; for now assume all samples have the same n_trials
```

Comment:
Do you plan the change the assumption that n_trials is always the same?  Is there even a "standard" definition for multinomial hmms where n_trials can change?  (e.g. is n_trials also part of the state?)

Reply:
Originally I intended to allow this, but I'll have to look more carefully to see how big of a change to the code structure it would be, or if something like that would even be sensible. 

As to whether there's a "standard" definition for multinomial hmms, I don't know. But from my understanding, "multinomial HMM" = "an HMM whose emission probabilities are parameterized by multinomial distributions", so by that definition something like that should be allowed. However, implementation/practice-wise, it's not clear to me if that makes sense, especially if each sample to learn from has a different n_trials (and it would also be harder to check valid input samples).

I think it's also possible to have n_trials be different for each component, but keep them the same for all emissions of that component.